packages/realm/bindgen/js_spec.yml

```diff
@@ -17,3 +17,12 @@ classes:
       get_cpu_arch: () -> std::string
       # print: (const char* fmt, ...) # can't expose varargs directly. Could expose a fixed overload.
 
+  WeakSyncSession:
+    cppName: std::weak_ptr<SyncSession>
```

Comment:
This is a _bit_ of a hack, but I think it worked out reasonably well. I wanted to keep this contained to realm-js since I don't think other SDKs need to do this, and I'd rather not expose them to `weak_ptr` unless we have to. I also didn't want to have to start thinking about how we would handle `weak_ptr` generically in when generating code, so I kept it as a special case for SyncSession.